numpy/polynomial/polyutils.py

```diff
@@ -725,6 +726,182 @@ def _fit(vander_f, x, y, deg, rcond=None, full=False, w=None):
+           max_degree=None):
+    """A simple 2D polynomial fit to data x, y, z
+    The polynomial can be evaluated with
+    numpy.polynomial.polynomial.polyval2d
+
+    Parameters
+    ----------
+    vander2d_f : {function(array_like, ..., int) -> ndarray, list of function(array_like, int) -> ndarray}
+        The 2d vander function, such as ``polyvander2d``,
+        or a list of 1d vander functions for each dimension
+    coords : list of array_like
+        x, y, z, ... coordinates
+    data : array_like
+        data values
+    degree : {int, ndim-tuple, ndim-array}, optional
+        degree of the polynomial fit in each dimension, by default 1.
+        If just one int, all dimensions use the same fit degree.
+        If a 1d array, then it specifies the degrees in
+        each dimension seperately.
+        If nd-array, it should have dimensions 
+        i.e. like the output coefficients. With degrees that should be used in
+        the fit set to non-zero values.
+    max_degree : {int, None}, optional
+        if given the maximum combined degree of the coefficients is
+        limited to this value, by default None
+
+    Returns
+    -------
+    coeff : array of shape (deg+1, deg+1)
+        the polynomial coefficients in numpy 2d format,
+        i.e. coeff[i, j] for x**i * y**j
+    """
+    ndim = len(coords)
+    if ndim == 0:
+        raise ValueError("Need at least one dimension to fit")
+
+    # Convert input to floating point numpy arrays if necessary
+    coords = list(coords)
+    for i in range(ndim):
+        coords[i] = np.asarray(coords[i]) + 0.0
+    data = np.asarray(data) + 0.0
+
+    deg = np.asarray(deg)
+    if w is not None:
+        w = np.asarray(w) + 0.0
+
+    if deg.dtype.kind not in 'iu' or deg.size == 0:
+        raise TypeError("deg must be an int or non-empty array of int")
+    if ndim != 1 and deg.ndim == 1 and not deg.size == ndim:
+        raise ValueError(f"deg must be of length {ndim}, if it is a 1d array")
+    if deg.ndim == ndim and np.all(deg == 0):
+        raise ValueError(f"deg must have at least one non-zero value, if it is an {ndim} dimensional array")
+    if deg.ndim not in [0, 1, ndim]:
+        raise TypeError(f"deg must be an array of dimension 0, 1, or {ndim}")
+    if deg.min() < 0:
+        raise ValueError("expected all deg >= 0")
+    if data.ndim != 1 and data.ndim != ndim:
+        raise TypeError(f"expected 1D or {ndim}D array for data")
+    if data.size == 0:
+        raise TypeError("expected non-empty vector for data")
+    npoints = len(data)
+    for i, x in enumerate(coords):
+        if x.ndim != 1 and x.ndim != ndim:
+            raise TypeError(f"expected 1D or {ndim}D vector for coords[{i}]")
+        if x.size == 0:
+            raise TypeError(f"expected non-empty vector for coords[{i}]")
+        if len(x) != npoints:
+            raise TypeError(f"expected coords[{i}] and data to have the same length")
+    if w is not None:
```

Comment:
This is always `True` due to line 774. Did you mean to put line 774 after this point?

Reply:
line 774 is only called if w is not None ?

Reply:
Ah, my mistake.